src/proof_of_possession.rs

```diff
@@ -346,6 +335,52 @@ mod test {
         .unwrap();
     }
 
+    #[tokio::test]
+    async fn basic_didkey_p256() {
+        let expires_in = Duration::minutes(5);
+        let jwk = JWK::generate_p256();
+        let did = DIDKey::generate(&jwk).unwrap();
+        let did_url = DIDKey
+            .resolve(&did.to_owned())
+            .await
+            .unwrap()
+            .document
+            .verification_method
+            .first()
+            .unwrap()
+            .id
+            .clone();
+        let pop_jwt = ProofOfPossession::generate(
+            &ProofOfPossessionParams {
+                issuer: "test".to_string(),
+                audience: Url::parse("http://localhost:300").unwrap(),
+                nonce: None,
+                controller: ProofOfPossessionController {
+                    jwk,
+                    vm: Some(did_url.parse().unwrap()),
+                },
+            },
+            expires_in,
+        )
+        .to_jwt()
+        .unwrap();
+        let resolver: VerificationMethodDIDResolver<_, AnyMethod> = DIDKey.into_vm_resolver();
+        let pop = ProofOfPossession::from_jwt(&pop_jwt, resolver)
+            .await
+            .unwrap();
+        pop.verify(&ProofOfPossessionVerificationParams {
+            nonce: pop.body.nonce.clone(),
+            audience: pop.body.audience.clone(),
+            issuer: "test".to_string(),
+            controller_did: Some(did_url.to_owned()),
```

Comment:
`did_url` is already owned, no need to call `.to_owned()` here, it will make an unnecessary copy.